### PR TITLE
Use a docker image to generate protobufs

### PR DESCRIPTION
**Pull request by gregwebs**
Protobufs are tricky to install, especially for multiple languages.
But we already have a reliable install in our github workflows.

Build a docker image based of the github workflows.

This is tested to work for Go and Rust. For C++ it makes it through without error but it doesn't seem to be updating the generated code. Perhaps someone familiar with the C++ build can contribute on top of this PR.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,55 @@
+FROM ubuntu:20.04
```
Comment (BusyJay) [inline Dockerfile:1]: How about using namely/docker-protoc to save a lot of build time?

Comment (BusyJay) [reply Dockerfile:1]: And setting environment GRPCIO_SYS_USE_PKG_CONFIG to 1 will force rust to link to share library grpc too, which can also save build time further.

Comment (gregwebs) [reply Dockerfile:1]: I tried namely/docker-protoc previously and ran into complications. This PR matches our github workflows which made it relatively easy to get working.

Comment (gregwebs) [reply Dockerfile:1]: I tried adding the env variable you recommend and then the build errors with ` thread 'main' panicked at 'can't find library grpc via pkg-config`. I am quite happy with this PR as is but would welcome future PRs that improve things further.

Comment (gregwebs) [reply Dockerfile:1]: A way to save docker image build time the best way would be to publish this image on dockerhub. I don't know who to contact about that.

**Review by BusyJay — APPROVED** [review]

**Review by overvenus — APPROVED** [review]
LGTM

**Pull request merged**